src/toolkitAPI.c

```diff
@@ -995,6 +995,42 @@ int DLLEXPORT swmm_getSubcatchStats(int index, TSubcatchStats *subcatchStats)
 	return (errorcode);		
 }
 
+int DLLEXPORT swmm_getSubcatchBuildup(int index, TSubcatchBuildup *subcatchBuildup)
```

Comment:
@katmratliff, there is a function that the API user can call to fetch the units out of SWMM.  You might want to open up another type that returns whether SWMM is reporting kg or lb.  

```C
int DLLEXPORT  swmm_getSimulationUnit(int type, int *value)
//
// Input: 	type = simulation unit type
// Output: 	enum representation of units
// Returns: API Error
// Purpose: get simulation unit types
{
	// Check if Open
	if(swmm_IsOpenFlag() == FALSE) return(ERR_API_INPUTNOTOPEN);
	
	// Output unit
	switch(type)
	{
		// System Unit (enum.h UnitsType)
		case 0:  *value = UnitSystem; break;
		// Flow Unit (enum.h FlowUnitsType)
		case 1:  *value = FlowUnits; break;
		// Concentration Unit
		//case 2:  *value = UnitSystem; break;
		// Type not available
		default: return(ERR_API_OUTBOUNDS); 
	}
	
	return (0);
}
```

Reply:
If you do that, feel free to clean up this function a bit.

Reply:
@bemcdonnell I think by querying the unit system and figuring out if the units are US or SI, that will already indicate kg or lb -- do you think it's redundant to add in a separate case for kg/lb? If not, I'll work on adding that. Also, is there a reason the concentration units aren't available? That seems like something I should definitely get up and working.

Reply:
@katmratliff, I have some thoughts on this - I will hopefully get to this in more detail this afternoon.   In summary - I agree with you that the units query does not belong in this function I proposed above...  and I think there could be a separate function that is more specific to the pollutants.  

But before we dive deeper into a simple function to only fetch units, perhaps we should plan out the pollutants portion of the API.  Most likely this will extend far beyond the scope of this PR so we can keep it high-level for now.  :-)

If you could help me understand something, it might shine some light on the the future API for pollutants. (Also, please forgive my naivety in advance - I'm more in tune with conveyance vs water quality in SWMM).  In the additions you made `swmm_getSubcatchStats()` function call, you are returning total current mass built up across the land surfaces, correct? So this would be `lbs` or `kg` instead of `concentration`.  Which makes sense because accumulated mass is not a concentration.  

The reason I ask is because I think the `stats` functions should house totalizers of mass and different peak values.  On the flip side, there are a few `results` getters which hold instantaneous results (e.g. swmm_getSubcatchResult() -> has current runoff rate).  If you are interested in runoff mass concentration, it seems like a it would fit better in a the Result getters.

I believe everything is in the correct place for the work you are doing! But I just wanted to have a quick discussion on this. I'll step through the logic later today.

Reply:
@bemcdonnell I was also thinking it along similar lines about where unit info could reside (could also be an associated key in the stats dict). 

And yes, you are correct about the surface build up units being in lbs or kg! I was just thinking that it would be useful to be able to pull out the correct concentration units when looking at the pollutant loading (we use both count/L as well as mass concentrations in our work), so it would also be useful to be able to extract those units via pyswmm.

Following your logic, I think the surface buildup functions reside in the correct place.